AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -32,6 +32,17 @@ public protocol APIClientProtocol: AnyObject {
     
 }
 
+/// :nodoc:
+/// Describes any async API Client.
+@available(iOS 15.0.0, *)
+public protocol AsyncAPIClientProtocol: AnyObject {
+    
+    /// :nodoc:
+    /// Performs the API request asynchronously.
+    func perform<R: Request>(_ request: R) async -> Result<R.ResponseType, Error>
```

Comment:
Do we want this function to be interpreted by the conformer? As in can/should it have different implementations?

Reply:
this is purely so that we can hide the implementation. and it is the case for the regular API as well. i don't think anyone would implement their own real APIClient 🤔